src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -11855,6 +11855,93 @@ namespace WaterThermalTanks {
         }
     }
 
+    bool HeatPumpWaterHeaterNodeAFNExceptions(EnergyPlusData &state, int const NodeNumber)
+    {
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Jeremy Lerond & Yan Chen
+        //       DATE WRITTEN   January 2021
+        //       MODIFIED       na
+        //       RE-ENGINEERED  na
+
+        // PURPOSE OF THIS FUNCTION:
+        // Check if a node is used by a heat pump water heater
+        // and can be excluded from an airflow network.
+
+        // Return value
+        bool HeatPumpWaterHeaterNodeException; 
+
+        int HeatPumpWaterHeaterIndex;
+
+        if (state.dataWaterThermalTanks->getWaterThermalTankInputFlag) {
+            GetWaterThermalTankInput(state);
+            state.dataWaterThermalTanks->getWaterThermalTankInputFlag = false;
+        }
+
+        HeatPumpWaterHeaterNodeException = false;
+
+        for (HeatPumpWaterHeaterIndex = 1; HeatPumpWaterHeaterIndex <= state.dataWaterThermalTanks->numHeatPumpWaterHeater; ++HeatPumpWaterHeaterIndex) {
+
+            // Get heat pump water heater data
+            HeatPumpWaterHeaterData &HPWH = state.dataWaterThermalTanks->HPWaterHeater(HeatPumpWaterHeaterIndex);
+
+            // "Zone and outdoor air" configuration is expected to change the zone pressure balance
+            if (HPWH.InletAirConfiguration != AmbientTempEnum::ZoneAndOA) {
```

Comment:
The "Zone and outdoor air" configuration is expected to change the zone pressure balance so should probably not be modeled with an AFN.